### PR TITLE
Better wayland

### LINKED ISSUE #210: nvidia-settings doesn't work as intended on wayland
# Expected Behavior

The extension should work as it works on X11. Currently on X11 it displays GPU Utilization in percents, Temperature in Celsius, GPU memory usage in percents, fan speed and GPU wattage.

# Current Behavior

When selected `nvidia-settings` as a properties provider, following notification appears:

```
Execution of nvidia-settings -q GPUUtilization -q [GPU]/GPUCoreTemp -q UsedDedicatedGPUMemory -q TotalDedicatedGPUMemory -q GPUCurrentFanSpeed -t failed:: ERROR: Error resolving target specification '' (No targets match target specification), specified in query 'GPUUtilization'
```

# Failure Information

The failure is likely related to `nvidia-settings` support for Wayland, not a bug of the gnome extension itself. As trying to execute the query via cli works on X11 but not on Wayland.

```
nvidia-settings -q GPUUtilization -t

ERROR: Error resolving target specification '' (No targets match target specification), specified in query 'GPUUtilization'.
```

## Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Enable Wayland: `sudo nano /etc/gdm3/custom.conf`
2. Change to true `WaylandEnable=true`
3. Log out
4. Before logging back in, select `Pop with Wayland`.

## Context

* Nvidia-Settings Version: 510.47.03
* Nvidia-smi version (N/A): 525.105.17
* Operating System: Pop OS 22.04 LTS

## Failure Logs

```
$ journalctl -b 0 -g nvidia-settings
Apr 22 01:24:28 pop-os gnome-shell[24503]: error: Execution of nvidia-settings -q GPUUtilization -q [GPU]/GPUCoreTemp -q UsedDedicatedGPUMemory -q TotalDedicatedGPUMemory -q GPUCurrentFanSpeed -t failed:: ERROR: Error resolving target specification '' (No targets match target specification), specified in query 'GPUUtilization'.
```

### PR DESCRIPTION
**Pull request by kenohassler**
This small PR addresses wayland usability issues. It changes the (default) SettingsAndSmiProvider to source all measurements from nvidia-smi, which should remove error messages popping up non-stop in case nvidia-settings is not installed.
Furthermore, I improved the readme to clarify how to restart the shell and how to select the properties to be displayed in the menu bar.

closes #210 #208 #202

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,19 +5,25 @@
 A Gnome extension to show Nvidia GPU information
 (https://extensions.gnome.org/extension/1320/nvidia-gpu-stats-tool/)
 
-# Requirements
-nvidia-settings and nvidia-smi
+## Requirements
+The extension requires `nvidia-smi`, which should be included with the drivers.
+Additionally, `nvidia-settings` is supported as a data source.
```
Comment (derenv) [inline README.md:10]: also optimus (i.e. optirun) supported

File: README.md [hunk 2]
```diff
@@ -5,19 +5,25 @@
...
-# Installation from git
+## Installation from git
     git clone https://github.com/ethanwharris/gnome-nvidia-extension.git
     cd gnome-nvidia-extension
     make
     make install
 
-Use `alt + f2`, `r`, `Enter` to restart the gnome shell (if this is not allowed then log out and log back in)
-Enable the extension in the gnome tweak tool.
+Restart gnome shell by logging out and back in.
```
Comment (derenv) [inline README.md:18]: Could be rephrased to specify wayland and avoid any ambiguity:
```
- restart gnome shell to load the installed extension:
    - On Wayland, log out and log back in
    - On X11, (etc)
- Afterwards (etc)
```

File: src/nvidiautil@ethanwharris/settingsAndSmiProvider.js
Comment (derenv) [inline src/nvidiautil@ethanwharris/settingsAndSmiProvider.js:1]: Is this worth changing? settingsProvider and smiProvider exists, a better option might be to remove this provider in its entirety - which should probably be it's own PR.

Comment (derenv) [reply src/nvidiautil@ethanwharris/settingsAndSmiProvider.js:1]: ah, forgot about the button! marking as resolved.

Comment (kenohassler) [reply src/nvidiautil@ethanwharris/settingsAndSmiProvider.js:1]: I think you have a point though. Why not integrate the button into smiProvider? It won't hurt, and then the settingsAndSmiProvider can be deleted. This should perhaps be a new PR indeed

Comment (derenv) [reply src/nvidiautil@ethanwharris/settingsAndSmiProvider.js:1]: > This should perhaps be a new PR indeed

I think so! The `provider` interface should probably be replaced at some point, honestly - it often seems like a bit of an unnecessary user interaction that could be confusing to some. But, any re-work runs into the issue of different distros packaging the driver, nvidia-settings and nvidia-smi differently - it may be an idea to just make them *both* a hard requirement for users (can't really do dependancies for extensions i think) and create some output when one of them isn't detected.

Comment (kenohassler) [reply src/nvidiautil@ethanwharris/settingsAndSmiProvider.js:1]: It wouldn't even be a hard requirement - if nvidia-settings is not installed and you press that button, there'll be a push message telling you nvidia-settings is not installed. It's not needed for anything else than opening with that button 😁

Comment (derenv) [reply src/nvidiautil@ethanwharris/settingsAndSmiProvider.js:1]: fair enough, that sounds reasonable!

File: README.md (hunk 1 above)
Comment (kenohassler) [reply README.md:14]: I decided not to advertise Bumblebee support explicitly as I'm unsure if this is still relevant ([the Arch wiki](https://wiki.archlinux.org/title/Bumblebee) disrecommends it). Should I still add a sentence here?

File: README.md (hunk 2 above)
Comment (kenohassler) [reply README.md]: done!

File: README.md (hunk 1 above)
Comment (derenv) [reply README.md:14]: You could while also mentioning that as a disclaimer?

Comment (derenv) [reply README.md:14]: > I decided not to advertise Bumblebee support explicitly as I'm unsure if this is still relevant ([the Arch wiki](https://wiki.archlinux.org/title/Bumblebee) disrecommends it). Should I still add a sentence here?

You could while also mentioning that as a disclaimer?

Comment (derenv) [reply README.md:14]: Try this:
"NOTE: There is also the option to use optimus (i.e. optirun) as a data source - but [this isn't recommended](https://wiki.archlinux.org/title/Bumblebee) and has not been tested with a recent version. The feature will likely be deprecated in a future release."

**Review by derenv — APPROVED** [review]
